### PR TITLE
test: use SimpleBuffer for testing ISBReaderOrchestrator 

### PR DESCRIPTION
**Pull request by vigith**
Use SimpleBuffer to do more edge case testing for ISBReaderOrchestrator

### CODE REVIEW

File: rust/numaflow-core/src/pipeline/isb/simplebuffer.rs [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+            ),
+            Offset::String(str_offset) => {
+                // For testing, we use sequence as the first part of the string
+                let seq: i64 = std::str::from_utf8(&str_offset.offset)
+                    .ok()
+                    .and_then(|s| s.parse().ok())
+                    .unwrap_or(0);
+                numaflow_testing::simplebuffer::Offset::new(seq, str_offset.partition_idx)
+            }
+        };
```
Comment (vaibhavtiwari33) [inline rust/numaflow-core/src/pipeline/isb/simplebuffer.rs]: nit: Should we implement `From` for `impl From<&Offset> for numaflow_testing::simplebuffer::Offset`? 
I see that we already have `impl From<&numaflow_testing::simplebuffer::Offset> for Offset`

Mentioning since `nack` and `mark_wip` also use the same logic

File: rust/numaflow-core/src/pipeline/isb/simplebuffer.rs [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+    }
+}
+
+/// Test type configuration that uses SimpleBuffer for ISB operations.
+///
+/// This allows testing ISBReaderOrchestrator and ISBWriterOrchestrator
+/// without requiring external infrastructure like NATS.
+#[derive(Clone)]
+#[allow(dead_code)] // Will be used when ISBReaderOrchestrator/ISBWriterOrchestrator integration tests are added
+pub(crate) struct WithSimpleBuffer;
```
Comment (vaibhavtiwari33) [inline rust/numaflow-core/src/pipeline/isb/simplebuffer.rs]: nit: I see it being constructed in the reader

Comment (vigith) [reply rust/numaflow-core/src/pipeline/isb/simplebuffer.rs]: I have a local commit for this, will push it in.

File: rust/numaflow-core/src/pipeline/isb/simplebuffer.rs (hunk 1 above)
Comment (vigith) [reply rust/numaflow-core/src/pipeline/isb/simplebuffer.rs]: good catch

File: rust/numaflow-core/src/pipeline/isb/reader.rs [hunk 3]
```diff
@@ -1322,3 +1322,497 @@ mod tests {
...
+            fourth.is_err(),
+            "4th message should block due to backpressure"
+        );
+
+        // Ack first message to free a permit by taking it out and dropping
+        let first_msg = inflight.remove(0);
+        drop(first_msg);
+
+        // Small delay to allow the wip_loop to complete ack and release permit
+        sleep(Duration::from_millis(100)).await;
```
Comment (yhl25) [inline rust/numaflow-core/src/pipeline/isb/reader.rs]: This if flaky!

File: rust/numaflow-core/src/pipeline/isb/reader.rs [hunk 4]
```diff
@@ -1322,3 +1322,497 @@ mod tests {
...
+                &reader,
+                &offset,
+                &cancel_clone,
+            )
+            .await
+        });
+
+        // Give it time to start retrying
+        sleep(Duration::from_millis(50)).await;
+
```
Comment (yhl25) [inline rust/numaflow-core/src/pipeline/isb/reader.rs]: flaky!

**Review by yhl25 — APPROVED** [review]
